### ISSUE TITLE
Battle Garegga Checksum Loop ROM0 Bad

### ISSUE THREAD

**Issue by hyp36rmax** [psomashekar/pr4m0d-MISTer-bin#19]
Battle Garegga Checksum loops endlessly with ROM0 Bad. Same rom used with 4 other misters with no issue.

**Comment by retrorepair** [comment]
Same issue here. Tried lots of MAME ROM versions with no change. Is there a specific version required or is .245 ok?

**Comment by hyp36rmax** [comment]
> Same issue here. Tried lots of MAME ROM versions with no change. Is there a specific version required or is .245 ok?

Are you playing on just Mister or with an addon such as Mistercade? At the moment i'm not sure if it's my setup. Depending on your setup we can confirm it's the core.

**Comment by KidpandaAG** [comment]
I am using a mistercade, game loads fine.

**Comment by hyp36rmax** [comment]
> I am using a mistercade, game loads fine.

Lol I have two addition mistercades which boots fine also. Maybe it’s the sdram Modules and timing on the core???

**Comment by retrorepair** [comment]
It may well be. It seems reloading the core has different results, sometimes ROM0 is good and ROM1 is bad, sometimes the other way round, sometimes both are bad and once both were ok and it booted the game!

**Comment by retrorepair** [comment]
I've not had issues with any other cores with my SDRAM (128mb) bit happy to run some tests to see if that's why.

**Comment by fraggore** [comment]
Can confirm when loading Garegga i get the bad rom error, not using mistercad just a standard mister setup, it's fine on 2 other mister setups i have tried it on.

Music is to fast in the 2 Mahou games Sorcer Striker is missing some sound effects like on the first boss just before he uses the fire attack there is a sound its missing on the core Kingdom Grandprix is missing some as well.

**Comment by retrorepair** [comment]
Must be a SDRAM issue. Could you run the speed test on the working misters and then on the one it doesn't work on? I'll do this in a bit and will see if I can verify it's the ram's safe operating speed that's causing issues.

I'd be surprised if this system requires ram that's *that* fast. If it does though, I think the v2.9 sdram may be required as I believe it fixes issues those with v2.5 had.

**Comment by fraggore** [comment]
it's the SDram test you want me to run yeah ?.

**Comment by retrorepair** [comment]
Yeah if you can

**Comment by fraggore** [comment]
ok running on mine now the other one is mate mister soon as i can get hold of him i will get him to run the test and post the results, he my have gone out for the day tho we will see.

**Comment by retrorepair** [comment]
No rush, I won't be able to do mine for an hour or two

**Comment by fraggore** [comment]
juat to let you know if this makes a diffrent Sorcer Striker and Kingdom Grandprix work for him just Garegga get the bad rom error.

**Comment by hyp36rmax** [comment]
You guys try swapping sdram modules?

**Comment by fraggore** [comment]
cant really do that as he lives about 10 miles away unfortunely

**Comment by retrorepair** [comment]
Well I've confirmed all the other games on this core are fine, just garegga has random rom check errors. I say random, it's only the same on every retry. A core reset or reload tends to change which rom fails. This time again, both passed and the game loaded.

It seems I can't twst the sdram on this monitor either as it won't sync to my display 🙄

**Comment by fraggore** [comment]
ram test on my mister that Garegga is working on has been running for an hour with no errors

**Comment by hyp36rmax** [comment]
> cant really do that as he lives about 10 miles away unfortunely

Sorry I meant swap sdram modules (if you have multiple misters)

**Comment by retrorepair** [comment]
> ram test on my mister that Garegga is working on has been running for an hour with no errors

At 130mhz? Probably more interesting on a mister that won't run it.

I don't have another ram to test sadly

**Comment by hyp36rmax** [comment]
Here’s my current test with system with failing Garegga I’m also having issues with bad rom on Kingdom Grand Prix on same setup. 
![Uploading 17E27ED3-F668-4F29-9783-FB3ED020A150.jpeg…]()

**Comment by fraggore** [comment]
My mates tested the New Gaegga core and it now boots fine thanks for that and the fast fix, i was giving the mahou games a good go last night and the sounds effects are there just at a very low level

**Comment by buddorz** [comment]
Not sure if this is the thread to add my findings, but they seem similar.

I've tested with the Garegga core - Battle Garegga and Sorcer Striker, and I've had some weird issues.

Intermittently, the game will load, but sometimes I will get the ROM0 or ROM1 issue. 
I deleted the roms and the mra/rbf files and ran update_all from today (July 29th), but still see the issue.

In Garegga:
If the game loads all the way, about 10-15 seconds after getting into a new game (or in the stage 1 demo if I let it loop), the game will soft reset the CPU and do the ram/rom/memory checks, while the music keeps playing. The game will boot and then reset after getting into the demo loop. Eventually, the main CPU crashes and the game display doesn't update, but the music will continue to play and loop. 

In Sorcer Striker:
Sometimes during gameplay, the game will flash black, pause, and load the region disclaimer, then go back to gameplay, except the background will be black, but the stage background will load once the game starts scrolling again. When this happened, I was playing as Chitta (not sure if it's character specific, but I couldn't reproduce it with Gain).

I tried to reproduce it again to get screenshots, but yeah, it's intermittent at least in Sorcer Striker. Battle Garegga is unplayable at all for me. 

My MiSTeR setup is one from Mister Addons, probably around 2019, with a 128MB SDRAM module version 2.5. I tested the SDRAM and it seems stable at 140mhz, at least for the 10 minute test.
